File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length;
        int repeated = 0;
        int missing = 0;

        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                int x = grid[i][j];
                map.put(x, map.getOrDefault(x, 0) + 1);
            }
        }

        for (int i = 1; i <= n * n; i++) {
            int count = map.getOrDefault(i, 0);

            if (count == 2) {
                repeated = i;
            } else if (count == 0) {
                missing = i;
            }
        }

        return new int[]{repeated, missing};
    }
}

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        ls=[-1]*(len(grid)*(len(grid[0])+1))
        for i in range(0,len(grid)):
            for j in range(0,len(grid)):
                ls[grid[i][j]]=ls[grid[i][j]]+1
        ls1=[-1]*2
        print(ls)
        for i in range(1,len(ls)):
            if ls[i]==-1:
                ls1[1]=i
                break
        for i in range(1,len(ls)):
            if ls[i]>=1:
                ls1[0]=i
                break
        return ls1
